### PR TITLE
Pootle data signals and data migration

### PR DESCRIPTION
**Pull request by phlax**
This PR will add

- signal handlers to ensure that stats are updated at the right times
- signal suppression at relevant parts of the code to ensure that stats generation is efficient
- migration to generate stats from existing data

### CODE REVIEW

File: pootle/apps/pootle_store/updater.py [hunk 1]
```diff
@@ -213,7 +214,10 @@ def increment_unsynced_unit_revision(self, update_revision):
         units = self.target_store.unit_set.filter(**filter_by)
         count = units.count()
         if count:
-            units.update(revision=Revision.incr())
+            # we update after here to trigger a stats update
+            # for the store after doing Unit.objects.update()
+            with update_data_after(self.target_store):
+                units.update(revision=Revision.incr())
```
Comment (ta2-1) [inline pootle/apps/pootle_store/updater.py]: I'm not sure that stats update is triggered here.

Comment (phlax) [inline pootle/apps/pootle_store/updater.py]: how so?

Comment (ta2-1) [inline pootle/apps/pootle_store/updater.py]: I misunderstood how and where we send `update_data

File: pootle/apps/accounts/utils.py [hunk 2]
```diff
@@ -165,44 +168,52 @@ def purge(self):
...
-        logger.debug("Deleting remaining submissions for: %s", self.user)
-        self.user.submission_set.all().delete()
-
-        # Delete remaining suggestions.
-        logger.debug("Deleting remaining suggestions for: %s", self.user)
-        self.user.suggestions.all().delete()
+        stores = set()
+        with keep_data():
+            result = self.remove_units_created()
+            stores |= result
```
Comment (unho) [inline pootle/apps/accounts/utils.py]: `stores |= self.remove_units_created()` ?

File: pootle/apps/accounts/utils.py [hunk 3]
```diff
@@ -224,13 +235,16 @@ def revert_units_commented(self):
...
+        return stores
 
     @write_stdout(" * Reverting units edited by: %(user)s... ")
     def revert_units_edited(self):
         """Revert unit edits made by a user to previous edit.
         """
+        stores = set()
         # Revert unit target where user is the last submitter.
         for unit in self.user.submitted.iterator():
+            stores.add(unit)
```
Comment (unho) [inline pootle/apps/accounts/utils.py]: `unit`?

Comment (phlax) [inline pootle/apps/accounts/utils.py]: i think this is correct

File: pootle/apps/accounts/utils.py (hunk 2 above)
Comment (phlax) [inline pootle/apps/accounts/utils.py]: yep

File: pootle/apps/accounts/utils.py (hunk 3 above)
Comment (phlax) [inline pootle/apps/accounts/utils.py]: you are right

**Comment by unho** [comment]
Issues, nits and lack of tests aside, lgtm.

**Pull request merged**